### PR TITLE
Implementing admin ability to edit patient and mhwp info and split name into first name, last name

### PR DESCRIPTION
**Pull request by rheahemmadi**
I made quite a lot of changes so please switch to my branch and test a lot in case you find any bugs/things that may impact functionality you've changed.

Key Features:
**1. AdminService: Edit User Information Functionality**
- Added a new function edit_user_information to the AdminService class.
- The function allows the Admin to edit the first name, last name, email, and emergency contact information (emergency contact only for patients)]
- User prompts for editing user details, with conditional prompts based on the role (e.g., emergency contact is only relevant for patients).
- Added logic to display current user information to the Admin before any edits are made.

**2. PatientService Updates**
- Updated the edit_personal_information function in PatientService to work with first name, last name,

**3. Data Handling in data_utils.py (Decode User Function)**
- Updated the decode_user function to correctly parse user information (first name, last name, email, etc.) for different user roles.

**4. Updated the Admin, Patient and MHWP classes to include fields for first name, last name, and email.**
- Added getters and setters for new attributes
- Updated the to_dict method, ensuring all relevant fields are saved and loaded correctly from JSON.
Registration
Updates to patient personal info
Admin updating patient info
Admin updating mhwp info

### CODE REVIEW

**Review by minjnlgc — CHANGES_REQUESTED** [review]
Great job and well done! Just a few small suggestions:

I noticed a bug that occurs when updating other users' info, where all admins' emails are set to null. This happens because `decode_user` doesn't include the code to read the admin’s email, so the admin object created from `data.json` is missing the email attribute. As a result, when saving, it reads an empty email and writes that back to `data.json`.

Also, since you’ve added an email attribute to all user types, it might make sense to move this attribute to the parent User class to take advantage of inheritance and keep things organised in an OOP-friendly way.

**Review by ben-jthompson — DISMISSED** [review]
Looks good, I encountered the same bug that Minjing got with the admin emails resetting to null. I would just make one suggestion to use the clear_screen() utility function periodically so that the user interface gets updated (eg. making sure that the admin dashboard gets cleared before you display the interface for editing data). I would probably clear the screen at the point at which the user presses E on the dashboard, as well as just before you show the updated info. Otherwise, I think it looks good; I'm not sure how the rest of the functionalities are currently looking, but perhaps we keep the Admin ascii art as a constant on the page and load it directly after we clear the screen every time.

File: breeze/models/admin.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
 from .user import User
 
 class Admin(User):
-    def __init__(self, username, password, first_name = None, last_name = None, is_disabled=False):
+    def __init__(self, username, password, first_name = None, last_name = None, email= None, is_disabled=False):
```
[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/models/admin.py:4]: Should I also move first name and last name in that case?

[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/models/admin.py:4]: I moved first and last name too

File: breeze/services/admin_service.py [hunk 2]
```diff
@@ -51,7 +58,64 @@ def allocate_patient_to_mhwp(self):
...
 
     def edit_user_information(self):
-        pass
+        """
+        Allows admin to edit information for a patient or an MWHP
+        """
+        print_system_message("Edit User Information")
+
+        data = load_data(self.data_file_path)
+        users = data.get("users", [])
```
[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/services/admin_service.py:65]: removed - will show you code tomorrow in meeting. Seems to work when i test but syntax highlighting is a little odd

File: breeze/services/admin_service.py [hunk 3]
```diff
@@ -51,7 +58,64 @@ def allocate_patient_to_mhwp(self):
...
+        
+            user = next((u for u in users if u.get_username() == username), None)
+            if user:
+                print(f"Editing information for user: {user.get_username()}")
+                print("\nEnter the new information or leave blank to keep the current value:")
+            # get user input and update that user info
+                updated_first_name = input("First name: ").strip()
+                updated_last_name = input("Last name: ").strip()
+                updated_email = input("email: ").strip()
+                updated_emergency_contact_email = input("emergency contact email: ").strip()
```
[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/services/admin_service.py]: Done

File: breeze/utils/data_utils.py [hunk 4]
```diff
@@ -35,6 +35,8 @@ def decode_user(user_data):
          return Admin(
             username=user_data.get("username"),
             password=user_data.get("password"),
+            first_name=user_data.get("information", {}).get("firstName"),
+            last_name=user_data.get("information", {}).get("lastName"),
             is_disabled=user_data.get("isDisabled", False)
```
[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/utils/data_utils.py:40]: added

File: breeze/services/admin_service.py [hunk 5]
```diff
@@ -51,7 +58,64 @@ def allocate_patient_to_mhwp(self):
...
+        users = data.get("users", [])
+       
+        while True:
+            username = input("Enter the username of the user to edit: ").strip()
+
+            if not username:
+                print("Username cannot be empty. Please enter a valid username.")
+                continue
+        
+            user = next((u for u in users if u.get_username() == username), None)
```
[the remark this answers is not in the document]
Comment (rheahemmadi) [reply breeze/services/admin_service.py:74]: done, will show you the syntax highlighting tomorrow. It's working when I test for now

**Comment by rheahemmadi** [comment]
@ben-jthompson liked the clear screen suggestion, have implemented.

**Comment by rheahemmadi** [comment]
Think I have made all the changes so can re-review before merging

**Review by minjnlgc — DISMISSED** [review]
It works well for me!

**Review by abbieDuan87 — DISMISSED** [review]
Great work!

**Review by abbieDuan87 — APPROVED** [review]

**Pull request merged**